tasks/release.py

```diff
@@ -293,14 +293,34 @@ def tag_devel(ctx, release_branch, commit="HEAD", push=True, force=False):
 
 
 @task
-def finish(ctx, release_branch, upstream="origin"):
+def finish(ctx, release_branch, release_date=None, upstream="origin"):
     """Updates the release.json file for the new version.
 
+    Args:
+        release_branch: The Git branch from which the release is being finalized.
+            This branch should correspond to the release line you are finishing
+            (for example, "7.69.x"). It is used to determine the major version,
+            update module dependencies, and generate the release artifacts.
+        release_date: Date when the release was done. Expected format YYYY-MM-DD,
+            like '2025-09-03'. (default: today's date)
+        upstream: The name of the remote repository to push the finalized release
+            branch to. This is typically "origin", but can be changed if working
+            with a fork or a differently named remote. (default: "origin")
+
     Updates internal module dependencies with the new version.
     """
-
     # Step 1: Preparation
 
+    # Validate release_date (if provided)
+    try:
+        if release_date:
+            datetime.strptime(release_date, "%Y-%m-%d")
+    except ValueError as err:
+        raise Exit(
+            color_message(f"Invalid date `{release_date}`. Date should be valid and in format YYYY-MM-DD.", "red"),
+            code=1,
+        ) from err
```

Comment:
It's not super relevant here, but I'd would have the `try/except` block inside the `if` statement, it can't break if we do not reach the `strptime` call